website/docs/presets/gql-tag.md

```diff
@@ -0,0 +1,104 @@
+---
+id: gql-tag
+title: gql-tag
+---
+
+This preset generates typings for your inline `gql` function usages, without having to manually specify import statements for the documents. All you need to do is import your `gql` function and run codegen in watch mode.
+
+Huge thanks to [Maël Nison](https://github.com/arcanis), who conceptualized the foundation for this preset [over here](https://github.com/arcanis/graphql-typescript-integration).
+
+```ts
+import { gql } from '@app/gql';
+
+// TweetFragment is a fully typed document node
+const TweetFragment = gql(/* GraphQL */ `
+  fragment TweetFragment on Tweet {
+    id
+    body
+  }
+`);
+
+const TweetsQueryWithFragment = gql(/* GraphQL */ `
+  query TweetsWithFragmentQuery {
+    Tweets {
+      id
+      ...TweetFragment
+    }
+  }
+`);
+```
+
+{@import ../generated-config/gql-tag-preset.md}
+
+## Getting Started
+
+In order to use this preset, you need to add the following configuration to your `codegen.yml`:
+
+```yml
+schema: src/path/to/your/schema.graphql
+documents:
+  - 'src/**/*.ts'
+  - '!src/gql/**/*'
+generates:
+  ./src/gql/:
+    preset: gql-tag-preset
```

Comment:
@n1ru4l the goal of the preset is to generate a `d.ts` near the operation file? 
I think maybe we can generate a single `d.ts` as well? without a preset? and use `declare module` with wildcards to add types to a specific identifier in a file?

Reply:
The file does not generate any d.ts files, only a graphql.ts and gql.ts file: https://github.com/dotansimha/graphql-code-generator/pull/6267/files#diff-a133eb90520d99b44b280c99cc17eee7369c9df5be1f731c22671d12adec06f8

No files are generated near the operation files, the goal is to not clutter the repository with codegen artifact files